### PR TITLE
generation1 vm support

### PR DESCRIPTION
**Pull request by hvedati**

### CODE REVIEW

File: rpc/nodeagent/storage/virtualharddisk/moc_nodeagent_virtualharddisk.proto [hunk 1]
```diff
@@ -15,6 +15,12 @@ enum VirtualHardDiskType {
 	DATADISK_VIRTUALHARDDISK = 1;
 }
 
+enum HyperVGenerationType {
```
Comment (manoj-kadam) [inline rpc/nodeagent/storage/virtualharddisk/moc_nodeagent_virtualharddisk.proto:18]: Code is not aligned properly

**Comment by madhanrm** [comment]
This PR is still marked as draft. Let me know when to review this

File: rpc/cloudagent/compute/galleryimage/moc_cloudagent_galleryimage.proto [hunk 2]
```diff
@@ -59,6 +59,7 @@ message GalleryImage {
 	string sourcePath = 11;
 	Tags tags = 12;
 	ImageSource sourceType = 13;
+	HyperVGeneration hyperVGeneration = 14;
```
Comment (madhanrm) [inline rpc/cloudagent/compute/galleryimage/moc_cloudagent_galleryimage.proto:62]: Please make sure that the default is V2 and that behavior is maintained.

File: rpc/common/moc_common_common.proto [hunk 3]
```diff
@@ -160,3 +160,9 @@ enum ImageSource {
         HTTP_SOURCE  = 2;
         CLONE_SOURCE = 3;
 }
+
+enum HyperVGeneration {
+	HyperVGenerationV1 = 0;
```
Comment (madhanrm) [inline rpc/common/moc_common_common.proto:165]: Should we flip this to keep V2 as default

Comment (hvedati) [reply rpc/common/moc_common_common.proto:165]: in moc-sdk and wssd-sdk I make sure the default is V2. https://github.com/microsoft/moc-sdk-for-go/pull/96/files and https://github.com/microsoft/wssd-sdk-for-go/pull/187/files

**Review by madhanrm — APPROVED** [review]
Good to merge

**Comment by madhanrm** [comment]
can you resolve the conflicts?

**Review by madhanrm — DISMISSED** [review]

**Comment by madhanrm** [comment]
Good to merge

**Review by madhanrm — APPROVED** [review]
Good to merge

**Pull request merged**